### PR TITLE
fix: blur radius not working in expo-image

### LINKED ISSUE #29536: [expo-image][iOS] blurRadius not working anymore
### Minimal reproducible example

https://github.com/RRaideRR/expo-image-blurradius

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

98% confident, that it worked in the past with SDK50 (not sure if it worked with SDK51 though). On Android, the blurRadius is applied to the image, on iOS it is not.

Android
Bildschirmfoto 2024-06-07 um 14 32 58

iOS
Bildschirmfoto 2024-06-07 um 14 34 47

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.20.2 - ~/.nvm/versions/node/v18.20.2/bin/node
      Yarn: 1.22.19 - ~/.yarn/bin/yarn
      npm: 10.5.0 - ~/.nvm/versions/node/v18.20.2/bin/npm
      Watchman: 2024.04.29.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.15.2 - /opt/homebrew/opt/ruby/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.5, iOS 17.5, macOS 14.5, tvOS 17.5, visionOS 1.2, watchOS 10.5
    IDEs:
      Android Studio: 2023.3 AI-233.14808.21.2331.11709847
      Xcode: 15.4/15F31d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.11 => 51.0.11 
      react: 18.2.0 => 18.2.0 
      react-native: 0.74.2 => 0.74.2 
    npmGlobalPackages:
      eas-cli: 9.1.0
    Expo Workflow: managed

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check for issues with metro config
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK

### PR DESCRIPTION
**Pull request by vonovak**
# Why

closes #29536

NOTE: For iOS, after installing this fix, you may need to increase the `blurRadius` prop value to get the same effect as previously.

# How

Reason why blurRadius does not work now:

[This variable](https://github.com/SDWebImage/SDWebImage/blob/b8523c1642f3c142b06dd98443ea7c48343a4dfd/SDWebImage/Core/SDImageCacheDefine.m#L95) evaluates to false currently, causing [this branch](https://github.com/SDWebImage/SDWebImage/blob/b8523c1642f3c142b06dd98443ea7c48343a4dfd/SDWebImage/Core/SDImageCacheDefine.m#L108) to run.

blurRadius is done via transform [in Expo-image](https://github.com/expo/expo/blob/d9bdec0a2596437f38584df91133b9c8a37cdf1a/packages/expo-image/ios/ImageView.swift#L362)

but here [shouldTransformImage](https://github.com/SDWebImage/SDWebImage/blob/b8523c1642f3c142b06dd98443ea7c48343a4dfd/SDWebImage/Core/SDWebImageManager.m#L495) evaluates to false

```
(!originalImage.sd_isAnimated || (options & SDWebImageTransformAnimatedImage))
```
because `originalImage.sd_isAnimated` is true but `SDWebImageTransformAnimatedImage` flag is not present.

Because `shouldTransformImage` is false, the `SDImageBlurTransformer` transform is not applied.

There are (at least) 2 ways to fix this, but I'm not sure about their side-effects. I might be fixing this but breaking something else.

Option 1 (now in this PR): 

Enable `SDWebImageTransformAnimatedImage`. `SDImageBlurTransformer` is the only transform we use now, and it works for blurRadius, afaict. 

The SDWebImage sources give a bit of a warning - if we, in the future, need more transforms, they may not work because "transformers could not manage animated images".

```
    /**
     * We usually don't apply transform on animated images as most transformers could not manage animated images.
     * Use this flag to transform them anyway.
     */
    SDWebImageTransformAnimatedImage = 1 << 9,
```

Option 2:
forcing [the variable](https://github.com/SDWebImage/SDWebImage/blob/b8523c1642f3c142b06dd98443ea7c48343a4dfd/SDWebImage/Core/SDImageCacheDefine.m#L95) to evaluate to true by using [SDWebImageDecodeFirstFrameOnly](https://github.com/SDWebImage/SDWebImage/blob/b8523c1642f3c142b06dd98443ea7c48343a4dfd/SDWebImage/Core/SDWebImageDefine.h#L185). Then the output image will not be AnimatedImage and the transforms will be applied. 

However, as I was testing it, I can already tell that this breaks animations, so not a good approach.

# Test Plan

I'm adding more examples, with animations, to the bare-expo app. I'll push them later.

# Checklist

doesn't apply

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image/ios/ImageView.swift [hunk 1]
```diff
@@ -24,7 +24,10 @@ public final class ImageView: ExpoView {
 
   var loadingOptions: SDWebImageOptions = [
     .retryFailed, // Don't blacklist URLs that failed downloading
-    .handleCookies // Handle cookies stored in the shared `HTTPCookieStore`
+    .handleCookies, // Handle cookies stored in the shared `HTTPCookieStore`
+    // Images from cache are `AnimatedImage`s. BlurRadius is done via a SDImageBlurTransformer
+    // so this flag needs to be enabled. Beware most transformers cannot manage animated images.
+    .transformAnimatedImage,
```
[the remark this answers is not in the document]
Comment (vonovak) [reply packages/expo-image/ios/ImageView.swift:30]: 🤔 not a swift pro if it's just about style, I prefer [prettier's default](https://prettier.io/docs/en/options.html#trailing-commas) 😄

File: packages/expo-image/ios/ImageModule.swift [hunk 2]
```diff
@@ -50,7 +50,8 @@ public final class ImageModule: Module {
       }
 
       Prop("blurRadius") { (view, blurRadius: Double?) in
-        view.blurRadius = blurRadius ?? .zero
```
Comment (vonovak) [inline packages/expo-image/ios/ImageModule.swift:53]: <details>
  <summary>screenshots</summary>
Simulator Screenshot - iPhone 15 Pro Max - 2024-06-11 at 22 35 52
Simulator Screenshot - iPhone 15 Pro Max - 2024-06-11 at 22 35 17

</details>

**Comment by vonovak** [comment]
marked as ready to get feedback, I'll still add the examples

File: packages/expo-image/ios/ImageModule.swift (hunk 2 above)
Comment (tsapeta) [reply packages/expo-image/ios/ImageModule.swift:53]: Interesting 🤔 It looks more like the RN Image, so I'm fine with this change. Can you add a comment explaining why we're dividing the radius by 2?

**Review by tsapeta — APPROVED** [review]
Nice work! Thanks for fixing this and adding new examples 👍

**Review by alanjhughes — APPROVED** [review]

File: packages/expo-image/ios/ImageModule.swift (hunk 2 above)
Comment (vonovak) [reply packages/expo-image/ios/ImageModule.swift:53]: I'd love to give a better explanation but the truth is that 2 is a "magic number" in this case. https://developer.apple.com/documentation/coreimage/cigaussianblur is used on iOS to achieve the effect. It'd need some more digging to find the cause, prob not worth it :).

**Review by lukmccall — APPROVED** [review]
LGTM ✅

**Comment by sdmotameni** [comment]
Created a fresh expo project (SDK 51.0.28) and this issue persists. Where was this fix merged into?

**Comment by vonovak** [comment]
hello @sdmotameni @brandtnewlabs, the fix is available in expo-image 1.12.13 and later. 

see the note at the top:

> NOTE: For iOS, after installing this fix, you may need to increase the blurRadius prop value to get the same effect as previously.

Please increase try increasing the blur radius to a large value.

If it's not working for you, please open a new issue with a reproducer that we can verify.

I just tried this and it works as expected for me.

Thank you :)

**Pull request merged**